groups/nts/ntsu/ntsu_socketutil.cpp

```diff
@@ -2424,11 +2581,9 @@ ntsa::Error SocketUtil::receive(ntsa::ReceiveContext*       context,
         context->setEndpoint(endpoint);
     }
 
-#if defined(BSLS_PLATFORM_OS_LINUX)
-    if (wantTimestamp) {
-        extractTimestampsFromCtrlBlock(context, msg);
+    if (wantMetaData) {
+        control.decode(context, msg, options);
```

Comment:
I have a general concern and I don't know what to suggest as a solution. `receive` functions do not return any actual error if there is something wrong during decoding. And while it may be ok for RX timestamps, I think it's not ok for file descriptors transfer. And unfortunately there is no way to tell the user something like "in general data reception was performed correctly and you can use this data, but there also was an attempt to receive a control message and it failed". Maybe `receive` functions should return a tuple of `ntsa::Error`s?

Reply:
This is a good observation. However, `ReceiveControl::decode` doesn't actually have a failure mode, as currently implemented (and it has the same problem: we can "fail" to decode a file descriptor but successfully decode a timestamp.) An API that returns like an array of ntsa::Error sounds very unpleasant. For now, I think we should live with the (new) BSLS_LOG_WARN logging alert users that while reception of the data was successful, some reception of the meta-data was not.